main.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"runtime"
+
+	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
+	// to ensure that exec-entrypoint and run can make use of them.
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	"github.com/operator-framework/operator-lib/leader"
+	controllerconfig "github.com/pravega/bookkeeper-operator/pkg/controller/config"
+	v1beta1 "github.com/pravega/pravega-operator/api/v1beta1"
+	"github.com/pravega/pravega-operator/controllers"
+	"github.com/pravega/pravega-operator/pkg/version"
+	"github.com/sirupsen/logrus"
+	apimachineryruntime "k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	versionFlag bool
+	webhookFlag bool
+	log         = ctrl.Log.WithName("cmd")
+	scheme      = apimachineryruntime.NewScheme()
+)
+
+func init() {
+	flag.BoolVar(&versionFlag, "version", false, "Show version and quit")
+	flag.BoolVar(&controllerconfig.TestMode, "test", false, "Enable test mode. Do not use this flag in production")
+	flag.BoolVar(&webhookFlag, "webhook", true, "Enable webhook, the default is enabled.")
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
```

Comment:
Need to add schemes related to Pravegacluster type also
`utilruntime.Must(v1beta1.AddToScheme(scheme))`